src/Generation/MethodDetails.php

```diff
@@ -128,7 +138,7 @@ private static function maybeCreatePaginated(ServiceDetails $svc, MethodDescript
             if ($nextPageToken->isRepeated() || $nextPageToken->getType() !== GPBType::STRING) {
                 throw new \Exception("next_page_token field must be of type string.");
             }
-            if (!$resources->isRepeated() || ProtoHelpers::isMap($catalog, $resources)) {
+            if (!$resourceFieldValid) {
                 throw new \Exception("Item resources field must be a repeated field with field-number 1.");
```

Comment:
Maybe just have the "repeated" part, given the caveats above?

Reply:
Oops, thanks, forgot to update this message. Done.